service/student_service_test.go

```diff
@@ -0,0 +1,406 @@
+		t.Fatalf("expected NotFound error, got %v", err)
+	}
+}
+
+func TestStudentServer_GetStudent_DBError(t *testing.T) {
+	ctrl, mockRepo, mockTime, _, server := setupTest(t)
+	defer ctrl.Finish()
+
+	studentID := "some-id"
+
+	fixedTime := time.Date(2025, 7, 7, 12, 0, 0, 0, time.UTC)
+	mockTime.EXPECT().Now().Return(fixedTime).AnyTimes()
+
+	mockRepo.EXPECT().
+		GetByID(gomock.Any(), studentID).
+		Return(nil, errors.New("db error"))
+
+	_, err := server.GetStudent(context.Background(), &proto.GetStudentRequest{Id: studentID})
+
+	if err == nil {
+		t.Fatal("expected error, got nil")
+	}
+
+	st, ok := status.FromError(err)
+	if !ok || st.Code() != codes.Internal {
+		t.Fatalf("gRPC Internal error expected, but received: %v", err)
+	}
+}
+
+func TestStudentServer_UpdateStudent_Success(t *testing.T) {
+	ctrl, mockRepo, mockTime, mockTx, server := setupTest(t)
+	defer ctrl.Finish()
+
+	fixedTime := time.Date(2025, 7, 7, 12, 0, 0, 0, time.UTC)
+	mockTime.EXPECT().Now().Return(fixedTime).AnyTimes()
+
+	mockTx.EXPECT().WithTransaction(gomock.Any(), gomock.Any()).DoAndReturn(
+		func(ctx context.Context, fn func(context.Context) error) error {
+			return fn(ctx)
+		},
+	)
+
+	mockRepo.EXPECT().
+		GetByID(gomock.Any(), "some-id").
+		Return(&student.Student{
+			ID:        "some-id",
+			FirstName: "Dobby",
+			LastName:  "Surname",
+			Grade:     9,
+			CreatedAt: fixedTime,
+		}, nil)
+
+	mockRepo.EXPECT().Update(gomock.Any(), gomock.Any()).Return(nil)
+
+	req := &proto.UpdateStudentRequest{
+		Student: &proto.Student{
+			Id:        "some-id",
+			FirstName: "John",
+			LastName:  "Doe",
+			Grade:     10,
+		},
+	}
+
+	_, err := server.UpdateStudent(context.Background(), req)
+	if err != nil {
+		t.Fatalf("expected no error, got %v", err)
+	}
+
+}
+
+func TestStudentServer_UpdateStudent_DBError(t *testing.T) {
+	ctrl, mockRepo, mockTime, mockTx, server := setupTest(t)
+	defer ctrl.Finish()
+
+	fixedTime := time.Date(2025, 7, 7, 12, 0, 0, 0, time.UTC)
+	mockTime.EXPECT().Now().Return(fixedTime).AnyTimes()
+
+	mockTx.EXPECT().WithTransaction(gomock.Any(), gomock.Any()).DoAndReturn(
+		func(ctx context.Context, fn func(context.Context) error) error {
+			return fn(ctx)
+		},
+	)
+
+	mockRepo.EXPECT().
+		GetByID(gomock.Any(), "some-id").
+		Return(&student.Student{
+			ID:        "some-id",
+			FirstName: "Dobby",
+			LastName:  "Surname",
+			Grade:     9,
+			CreatedAt: fixedTime,
+		}, nil)
+
+	mockRepo.EXPECT().Update(gomock.Any(), gomock.Any()).Return(errors.New("db error"))
+
+	req := &proto.UpdateStudentRequest{
+		Student: &proto.Student{
+			Id:        "some-id",
+			FirstName: "John",
+			LastName:  "Doe",
+			Grade:     10,
+		},
+	}
+
+	_, err := server.UpdateStudent(context.Background(), req)
+
+	if err == nil {
+		t.Fatal("expected error, got nil")
+	}
+
+	st, ok := status.FromError(err)
+	if !ok || st.Code() != codes.Internal {
+		t.Fatalf("gRPC Internal error expected, but received: %v", err)
+	}
+}
+
```

Comment:
Do not forget to check business logic, for example - grade could not be decreased